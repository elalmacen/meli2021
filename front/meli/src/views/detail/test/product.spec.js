import React from 'react';
import {shallow} from 'enzyme';
import View from '../product';

describe('test detail/product view', () => {
    test('renders without crashing', () => {
        const categories = ['batman','robin'];
        const product = {
            id: "MLA850382509",
            title: "Lorem Impsum",
            picture: "http://http2.mlstatic.com/D_916062-MLA43654417337_102020-I.jpg",
            condition: "new",
            free_shipping: true,
            sold_quantity: 0,
            description: "Lorem Impsum",
            categories: ['batman','robin'],
            price: {
                currency: "ARS",
                amount: 185000,
                decimals: 0
            }
        };

        const wrapper = shallow(<View product={product} categories={categories} notFound={false} />);
        expect(wrapper).toMatchSnapshot();
    });    
});
