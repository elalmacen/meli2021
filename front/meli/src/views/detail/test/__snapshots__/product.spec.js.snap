// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test detail/product view renders without crashing 1`] = `
<Fragment>
  <BreadCrumb
    categories={
      Array [
        "batman",
        "robin",
      ]
    }
  />
  <Col
    className="product-view"
  >
    <Row
      noGutters={false}
    >
      <Col
        md={8}
        xs={12}
      >
        <img
          alt="thumbnail"
          src="http://http2.mlstatic.com/D_916062-MLA43654417337_102020-I.jpg"
        />
      </Col>
      <Col
        md={4}
        xs={12}
      >
        <div
          className="new"
        >
          Nuevo
           - 
          0 vendido
        </div>
        <h2
          className="product-title"
        >
          Lorem Impsum
        </h2>
        <CurrencyFormat
          allowNegative={true}
          className="price"
          decimalSeparator=","
          displayType="text"
          fixedDecimalScale={false}
          isAllowed={[Function]}
          isNumericString={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          onValueChange={[Function]}
          prefix="$ "
          suffix=""
          thousandSeparator="."
          thousandSpacing="3"
          type="text"
          value={185000}
        />
        <Button
          active={false}
          disabled={false}
          variant="primary"
        >
          Comprar
        </Button>
      </Col>
    </Row>
    <Row
      noGutters={false}
    >
      <Col
        md={12}
        xs={12}
      >
        <div
          className="description"
        >
          <h3>
            Descripci√≥n del producto
          </h3>
          <p>
            Lorem Impsum
          </p>
        </div>
      </Col>
    </Row>
  </Col>
</Fragment>
`;
